### PR TITLE
Add back-face culling to `ModelExperimental`

### LINKED ISSUE #10056: Back face culling for ModelExperimental
In `ModelExperimental`, we haven't yet added the `backFaceCulling` constructor option. This came up in #10055 when trying to remove `Gltf3DTileContent`. 

The ultimate goal is to set each DrawCommand's `renderState.cull.enabled` based on `material.doubleSided` and `model.backFaceCulling`. Though note that the latter can be changed at runtime, the former comes from the glTF model.

The way we do this in `ModelExperimental` will differ from the old `Model` because we're trying to avoid derived commands when possible.

This task involves:

- [ ] Add the `backFaceCulling` constructor option to `ModelExperimental` (and `.fromGltf()`)
- [ ] I'm not sure if the best place to put the initial configuration of the render state in `MaterialPipelineStage` or `buildDrawCommands()`. We might need to store the `doubleSided` property somewhere though in case `model.backFaceCulling` is toggled at runtime
- [ ] And a getter and setter in `ModelExperimental`. The setter will have to update the draw commands (but only if the value changed). This will involve iterating over the `ModelExperimentalSceneGraph` and updating the draw commands (from a quick glance it looks like they're stored per-primitive). 
- [ ] It might be helpful to make a function to iterate over the scene graph with a callback function (as other upcoming issues will also need to do similar updates with the draw commands)

### PR DESCRIPTION
**Pull request by j9liu**
Fixes #10056.

This PR adds the ability to toggle back-face culling for `ModelExperimental`, and by extension any tilesets containing glTF / glb content. It also provides a helper function to iterate over the runtime primitives in a `ModelExperimentalSceneGraph`, which will probably be relevant in the future.

@ptrgags @lilleyse could one of you take a look? also, should I log this as an addition or as a fix in `CHANGES.md`?

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @j9liu!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/modelexperimental-backfaceculling/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Review by ptrgags — COMMENTED** [review]
@j9liu this is looking good! Just a couple comments.

I also put together a quick [local Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=tVf/T+M2FP9XvP6UTtRJaGmhA7SjfDlOsJ2uvZu0dbqZxG0sHLuynVKY+N/3bCdpCkU77gQCx1+e3/u87yaRQhu0ZPSOKnSEBL1DI6pZkeMvbi+YthK3HklhCBNUTVvtX6ZiKvwdnHCZ3OKkUIoKM2E5BS4lhw8FZ0ScEkPxTMn8Usv9fhQHU4HQtLUb7cadOO5EB5PdaLg7GHYHOIr7Ub/XG+x1e/v9/d5Bd/DntDUVXl4YomuZUq6nInGgc7cCcX9Zjv/aASFDV2YI7D0EdEI4l1JMWzv+GC5SThMgmRUiMUwKFLSruwj5QycmqPYs2HC8oIkOQRUSehDhiVx1Tkhy2zknCe2MCs6ZmG/fxXNuZjUE+xPvVXOrm/0+ulM/PlFlfMuEoCkaZUSRxFgP/Jg2GIfwOyb5glOnkrdk6G12TUQ5+wpTwH6zAT16BfJTwCvRSOYLRbUGFRyWt4LvpK2FVeowfjtRBcToMy+8RpXPgjODwL3fhf5FzMDQca4nFUoUfZN9paBvGw1ORHP+LCBeFcsXShYiRV9oxhL+ZtC9lFLI5uqH4vkSyg4RCQTy9wbCN4AH1rWcjcXrysjfrmj6SpktntT295SkUJg+MpNknyT3wMqza2IybOQnoCBCB/F+1HZMIz/6cuz5ztiKpueK5HSigHYmVb6u/vWWxtAiCPd08ry+c0GhmxAjVdkRhFQmK/Wbtu6oNuviz2lZ76tW8J7NMw5/BpmMIriaE+gFFbCk0Ebm44ykT7vaqHESOB85LmAM56FhRXfV3MWff7u6nDhgM0XmOXQ6z2HiAuMfe7CULK1Pr6FTBufl4lIsCrCVdt8dxISEZfKQf3UKgbmpYoSjvJzUoTPjksA9qe6ISkGLigB7bc9G+KH0en2Sstms0LYF52wVLGnSDWIcQTEph/YOcptuHa83naigEnXsdn1IPdoBFHz0fqgjvBnVheI7KKPWYiV474WF1MwR1zExIsrAjIiuexGc0rmiVJdp0Yl3uzga9Hr9+KAM8l4PR3tRdxD1yw0vxc49HlQ+XbBOIJrwQrEcRC6pxormcknfQWx7RZyxAclL9CRNSxxVHtgLZ6sFWNZ6EaxrIV9ADgZ1btuMHCKrf7VT+VSx1XBLJmRPEq+ZD43qUFmukeuQw41VyfkM3hcLDZGH/7gY7/caBFtSs64VNdkNvFTsQ6V8p0ABgySi9XEzi4Ybq7LYtJuOcJqD2Ul6/xEMxTTFkJoiWBdFR1GHd/V+BIiKQImQt+9MbYJnBnipeBExbxpuew3r7bV3/o+mE28QeW1ubOcAWeNFRhXFCmgLjX5G67rrJ2Wu2M+jtYamkPWQk0vCA+gFR8de6ZLpptUhJn/aemCZQZZGUeRNPCYiTYg20MQgWCdS8huirqkovGG1p2rttA61uef0uML4K8sXUFptmAbQcQyFjgMFQ4c38CqiBidaV+3jMGxePUzZErH0aMu/ACjhRGs4mQHWMXuAVn58GAL9s6tQWawFf19Sxcm9Jcvi4yu/iTE+DGG5/abxGj7h/B8) to see this in action

**Comment by j9liu** [comment]
Thanks @ptrgags for the review! I meant to include the [sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=pVVtj+I2EP4rFp+CBE7C6y7HrtrlbrdX9XSrglqpyhfjTCBax0a2w7I97X+/sR0Ct6WnviBE8GQ888zzPE64ksaSfQnPoMkNkfBMFmDKuqK/+ViUdbhfL5S0rJSgs073XSYzGfZQLhR/orzWGqRdlRVglabCz7UomXzPLNBCq+qjUVeTJI0ySUjWGSSDtJ+m/eR6NUhmg+lsOKVJOkkmo9F0PBxdTa5G18PpH1knk6FfHJNPKgdBdsqUtlSylBty8+8+rg73Ex+LnNAumLb4j8mhR/seNhrAeLT9dDCkyXQ0mqTXPRcYjWgyTobTZOKXKR0HlKH2dveGyp+A5Yj2sbR8+6sSIkp6xH1PW4ryAPm9ZhWsNJOmULo6IWtDhiLZTIQ8dd/ueQDUhVmlG26l0nabdXph9QzGnmgM/SrHpMEWX1zOWh1m7uqy4+UOuIlRNRZ7vk18pw79O8af+veMQ39RC4HDXI7SjbCF7/waTMJ06IWropbcU25AALe+eFRr0Q0gGj8ZjsPQnS4r1GcPhmqo1B5+RNbcACSUQ+R/l8/yPArDNPT5Rh8OO8Ac9CgTXt8HBBp9CYmEONgzgmB6x8gaR3OTNYPNSMGEgfa2R/GJWV0eZhd02r5R/Fyt6FiDtC7snUJonrNVU/kDYtgZVeb094fl1egs4YJxjje7Tdpr112D+g1wJJXlL49IQ2mA2i3IqFUn8hmNKMd8roR6Y+qFC0UpdUZuf3AZVGr15AhNM1qIl5W6U7V0tCx3W9AQGtH1N8EeaTXJa/Q0Ipph6XE7sioKA3b23ePVPyrP7JY+fiQxGfgDN6Dj7pETj9JfXh0xSyZzzowV4Ay0UpuNgLvaWiXxCdh4AH3duICc2HJl+Bb4E+R4xAL6ZrBvHYTsNXmuZ5Dju129oKJGf9t/3Lksoib0HwVEqhtq3A+efvifRnhthz0/9uH5g9ofuu86vc7c2BcBt0eNfyirHT7C3HmMKI0tVDuBbxETr2sczVJuzNFl8/h86zwv96TMby68tAgXzBi8U6Acy/JPyDq38xjz/7JVKG+nz3vQgr24tG16+0sIUkrnMS4v77RKiTXTbyp/BQ) I made to test this when I opened the PR, sorry for not including at first. It also checks for the cull setting's interaction with translucency.

I followed your suggestions, just waiting on @lilleyse 's thoughts on our handling of the `DrawCommands`.

File: Source/Scene/ModelExperimental/ModelExperimentalSceneGraph.js [hunk 1]
```diff
@@ -409,6 +411,40 @@ ModelExperimentalSceneGraph.prototype.updateModelMatrix = function () {
...
+      const runtimePrimitive = runtimeNode.runtimePrimitives[j];
+      callback(runtimePrimitive);
+    }
+  }
+}
+
+/**
+ * Traverses through all draw commands and changes the back-face cull setting.
+ *
+ * @param {Boolean} backFaceCulling The new value for the back-face cull setting.
```
Comment (ptrgags) [inline Source/Scene/ModelExperimental/ModelExperimentalSceneGraph.js:427]: Just a tiny wording tweak
```suggestion
 * Traverses through all draw commands and changes the back-face culling setting.
 *
 * @param {Boolean} backFaceCulling The new value for the back-face culling setting.
```

File: Source/Scene/ModelExperimental/ModelExperimentalSceneGraph.js [hunk 2]
```diff
@@ -409,6 +411,35 @@ ModelExperimentalSceneGraph.prototype.updateModelMatrix = function () {
...
+ModelExperimentalSceneGraph.prototype.updateBackFaceCulling = function (value) {
+  const model = this._model;
+  forEachRuntimePrimitive(this, function (runtimePrimitive) {
+    for (let k = 0; k < runtimePrimitive.drawCommands.length; k++) {
+      const drawCommand = runtimePrimitive.drawCommands[k];
+      const renderState = clone(drawCommand.renderState, true);
+      const doubleSided = runtimePrimitive.primitive.material.doubleSided;
+      const translucent = defined(model.color) && model.color.alpha < 1.0;
+      renderState.cull.enabled = value && !doubleSided && !translucent;
+      drawCommand.renderState = RenderState.fromCache(renderState);
```
[the remark this answers is not in the document]
Comment (lilleyse) [reply Source/Scene/ModelExperimental/ModelExperimentalSceneGraph.js:443]: Yes, that should be fine.

**Comment by ptrgags** [comment]
Okay, now that we're all good with this approach to updating DrawCommands, I'm going to merge this.

Thanks @j9liu!

**Pull request merged**